core/modules/memman/MemFile.h

```diff
@@ -138,7 +140,10 @@ class MemFile {
     bool        _isMapped   = false;   // Protected by _fileMutex
     bool        _isReserved = false;   // Ditto
     bool        _isLocked   = false;   // Ditto
```

Comment:
So, _fileMutex protects _isMapped while _mlockFileMutex protects_ isLocked. Sounds like limited scope here can only one mutex be used for both or one mutex and an atomic?

Reply:
If _fileMutex gets held for a long period of time, it wedges the scheduler when it checks to see if a file has been mapped. This is terribly painful when a file that has already been mapped is blocked waiting for a long file lock, and this in turn blocks any other task (including interactive tasks) from being started. 

It looks like having 2 threads working their way through MemFile::memLock() at the same time would have some dangerous race conditions, so _mlockFileMutex keeps that from happening.